### ISSUE TITLE
Deadline extensions interfere with API submissions to unrelated modules

### ISSUE THREAD

**Issue by jsorva** [apluslms/a-plus#1282]
We have (at least) two documented cases where a student has been given a personal deadline extension to a module, but the student has then been unable to submit assignments to a _subsequent_ module. By “documented”, I mean that the students have complained very similarly about the symptoms; we haven’t been able to replicate the issue artificially.

Specifics: 
* Submissions through the web interface work fine. But the A+ API (used via the IntelliJ plugin) does not grant the student access.
* Here’s one student who was given a DL extension for Week 1 and Week 2, and was then unable to submit Week 3 assignments on schedule via the API: 
https://plus.cs.aalto.fi/o1/2023/teachers/participants/30359
* And here’s one student who was given a DL extension for Week 4, and is currently unable to submit Week 5 assignments on schedule via the API:
https://plus.cs.aalto.fi/o1/2023/teachers/participants/31290

Now, I’m not certain if this is an A+ bug or if it could have something to do with the IntelliJ plugin that our students use to submit via the API. But the symptoms somewhat suggest an A+ issue... and here’s another interesting detail: 

Both of those students’ Points pages (linked above) feature the display bug documented here: https://github.com/apluslms/a-plus/issues/1276. So, those pages don’t actually show that the students have extensions. (They do have extensions, though, as you can verify here, if you’re lucky enough to get this page to load: https://plus.cs.aalto.fi/o1/2023/teachers/deadline-deviations/ )
This seems potentially relevant, but I’m not sure what the connection is, if there is one.

**Comment by Mankro** [comment]
One thing to check is the cache and outdated/stale cached data just like in #1276.

**Comment by oseppala** [comment]
I encountered this bug yesterday with a student. The student was registered on the course and had a correct token. Other students were able to submit normally. the student also was able to submit exercises through the web interface.

Judging by the IJ plugin logs the plugin received a non-success status code from A-plus when requesting a list of submittable exercises through the API. I asked the student to check the situation after the deadline deviation passed (which was 1.5 hours from when I was shown the problem) . The student reported that everything worked correctly after the deviation passed.

**Comment by jsorva** [comment]
@Mankro @oseppala We have had a few more cases of this bug happening in the last few days. It’s very confusing for students.

Today, Otto and I checked the case that Otto mentioned above, and that student also had the https://github.com/apluslms/a-plus/issues/1276 symptom: no mention of the extensions on their Points page even though the extensions were in place.

Another student that I just chatted with had the same problem — submissions not working for the current module. However, for this student, https://github.com/apluslms/a-plus/issues/1276 does not apply: their extension to the previous module *is* visible on their Points page, but they are (according to their report, which I have no reason to doubt) unable to submit via the API.

**Comment by ihalaij1** [comment]
I looked into this, but was unable to reproduce the problem in my local development environment. I also looked at the cache code, but couldn't spot any obvious issues there.

We have some large cache changes still pending, so if the problem is indeed related to the cache, it might get "accidentally" fixed in the near future, when those changes go live.

**Comment by jsorva** [comment]
> We have some large cache changes still pending, so if the problem is indeed related to the cache, it might get "accidentally" fixed in the near future, when those changes go live.

OK, I hope so! This is continuing to affect some students.

**Comment by mikaelGusse** [comment]
@jsorva . The cache was changed after this issue was opened. Have you encountered this issue again over the past year?

**Comment by jsorva** [comment]
No, I don’t recall seeing this recently. Perhaps the problem has gone away.

**Comment by mikaelGusse** [comment]
Thanks! I'll close this issue then. If you encounter this again then we can open this issue again.
